Title is null

Crash from Google Play Console.
Application version: 104.
Android 4.4
```
java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.ReadArticleActivity}: java.lang.NullPointerException
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2391)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2442)
	at android.app.ActivityThread.access$800(ActivityThread.java:166)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1357)
	at android.os.Handler.dispatchMessage(Handler.java:110)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:5387)
	at java.lang.reflect.Method.invokeNative(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:515)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:831)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:647)
	at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:132)
	at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.NullPointerException
	at android.text.TextUtils.htmlEncode(TextUtils.java:1352)
	at fr.gaulupeau.apps.Poche.ui.ReadArticleActivity.getHtmlPage(ReadArticleActivity.java:614)
	at fr.gaulupeau.apps.Poche.ui.ReadArticleActivity.loadArticleToWebView(ReadArticleActivity.java:557)
	at fr.gaulupeau.apps.Poche.ui.ReadArticleActivity.onCreate(ReadArticleActivity.java:181)
	at android.app.Activity.performCreate(Activity.java:5363)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1088)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2335)
	... 12 more
```

Should probably check all values received from API and initialize variables with non-null values.

--- a/app/src/main/java/fr/gaulupeau/apps/Poche/network/Updater.java
+++ b/app/src/main/java/fr/gaulupeau/apps/Poche/network/Updater.java
@@ -209,73 +209,88 @@ public class Updater {
 
                 boolean existing = true;
                 if(article == null) {
-                    article = new Article(null);
-
                     existing = false;
+
+                    article = new Article(null);
+                    article.setArticleId(id);
+                    article.setTitle(apiArticle.title);
+                    article.setContent(apiArticle.content);
+                    article.setDomain(apiArticle.domainName);
+                    article.setUrl(apiArticle.url);
+                    article.setEstimatedReadingTime(apiArticle.readingTime);
+                    article.setLanguage(apiArticle.language);
+                    article.setPreviewPictureURL(apiArticle.previewPicture);
+                    article.setCreationDate(apiArticle.createdAt);
+                    article.setUpdateDate(apiArticle.updatedAt);
+                    article.setArchive(apiArticle.archived);
+                    article.setFavorite(apiArticle.starred);
+                    article.setImagesDownloaded(false);
 
                     articleChanges.add(ChangeType.ADDED);
                 }
 
                 if(existing) {
-                    if(!TextUtils.equals(article.getContent(), apiArticle.content)) {
+                    if(!equalOrEmpty(article.getContent(), apiArticle.content)) {
+                        article.setContent(apiArticle.content);
                         articleChanges.add(ChangeType.CONTENT_CHANGED);
                     }
-                    if(!TextUtils.equals(article.getTitle(), apiArticle.title)) {
+                    if(!equalOrEmpty(article.getTitle(), apiArticle.title)) {
+                        article.setTitle(apiArticle.title);
                         articleChanges.add(ChangeType.TITLE_CHANGED);
                     }
-                    if(!TextUtils.equals(article.getDomain(), apiArticle.domainName)) {
+                    if(!equalOrEmpty(article.getDomain(), apiArticle.domainName)) {
+                        article.setDomain(apiArticle.domainName);
                         articleChanges.add(ChangeType.DOMAIN_CHANGED);
                     }
-                    if(!TextUtils.equals(article.getUrl(), apiArticle.url)) {
+                    if(!equalOrEmpty(article.getUrl(), apiArticle.url)) {
+                        article.setUrl(apiArticle.url);
                         articleChanges.add(ChangeType.URL_CHANGED);
                     }
                     if(article.getEstimatedReadingTime() != apiArticle.readingTime) {
+                        article.setEstimatedReadingTime(apiArticle.readingTime);
                         articleChanges.add(ChangeType.ESTIMATED_READING_TIME_CHANGED);
                     }
-                    if(!TextUtils.equals(article.getLanguage(), apiArticle.language)) {
+                    if(!equalOrEmpty(article.getLanguage(), apiArticle.language)) {
+                        article.setLanguage(apiArticle.language);
                         articleChanges.add(ChangeType.LANGUAGE_CHANGED);
                     }
-                    if(!TextUtils.equals(article.getPreviewPictureURL(), apiArticle.previewPicture)) {
+                    if(!equalOrEmpty(article.getPreviewPictureURL(), apiArticle.previewPicture)) {
+                        article.setPreviewPictureURL(apiArticle.previewPicture);
                         articleChanges.add(ChangeType.PREVIEW_PICTURE_URL_CHANGED);
                     }
                     if(article.getCreationDate().getTime() != apiArticle.createdAt.getTime()) {
+                        article.setCreationDate(apiArticle.createdAt);
                         articleChanges.add(ChangeType.CREATED_DATE_CHANGED);
                     }
                     if(article.getUpdateDate().getTime() != apiArticle.updatedAt.getTime()) {
+                        article.setUpdateDate(apiArticle.updatedAt);
                         articleChanges.add(ChangeType.UPDATED_DATE_CHANGED);
                     }
                     if(article.getArchive() != apiArticle.archived) {
+                        article.setArchive(apiArticle.archived);
                         articleChanges.add(apiArticle.archived
                                 ? ChangeType.ARCHIVED
                                 : ChangeType.UNARCHIVED);
                     }
                     if(article.getFavorite() != apiArticle.starred) {
+                        article.setFavorite(apiArticle.starred);
                         articleChanges.add(apiArticle.starred
                                 ? ChangeType.FAVORITED
                                 : ChangeType.UNFAVORITED);
                     }
-                }
 
-                if(!existing || articleChanges.contains(ChangeType.CONTENT_CHANGED)) {
-                    if(article.getImagesDownloaded() != null && article.getImagesDownloaded()) {
-                        articleChanges.add(ChangeType.FETCHED_IMAGES_CHANGED);
+                    if(articleChanges.contains(ChangeType.CONTENT_CHANGED)
+                            || articleChanges.contains(ChangeType.PREVIEW_PICTURE_URL_CHANGED)) {
+
+                        if(article.getImagesDownloaded() != null && article.getImagesDownloaded()) {
+                            articleChanges.add(ChangeType.FETCHED_IMAGES_CHANGED);
+                        }
+
+                        article.setImagesDownloaded(false);
                     }
-
-                    article.setImagesDownloaded(false);
                 }
 
-                article.setArticleId(id);
-                article.setTitle(apiArticle.title);
-                article.setContent(apiArticle.content);
-                article.setDomain(apiArticle.domainName);
-                article.setUrl(apiArticle.url);
-                article.setEstimatedReadingTime(apiArticle.readingTime);
-                article.setLanguage(apiArticle.language);
-                article.setPreviewPictureURL(apiArticle.previewPicture);
-                article.setCreationDate(apiArticle.createdAt);
-                article.setUpdateDate(apiArticle.updatedAt);
-                article.setArchive(apiArticle.archived);
-                article.setFavorite(apiArticle.starred);
+                fixArticleNullValues(article);
 
                 List<Tag> articleTags;
                 if(existing) {
@@ -448,6 +463,28 @@ public class Updater {
         }
 
         return latestUpdatedItemTimestamp;
+    }
+
+    /**
+     * Returns true if both arguments are equal or empty.
+     * {@code null} and empty sequences are considered equal.
+     *
+     * @param s1 first char sequence
+     * @param s2 second char sequence
+     * @return true if arguments are considered equal
+     */
+    private boolean equalOrEmpty(CharSequence s1, CharSequence s2) {
+        return (TextUtils.isEmpty(s1) && TextUtils.isEmpty(s2))
+                || TextUtils.equals(s1, s2);
+    }
+
+    private void fixArticleNullValues(Article article) {
+        if(article.getTitle() == null) article.setTitle("");
+        if(article.getContent() == null) article.setContent("");
+        if(article.getDomain() == null) article.setDomain("");
+        if(article.getUrl() == null) article.setUrl("");
+        if(article.getLanguage() == null) article.setLanguage("");
+        if(article.getPreviewPictureURL() == null) article.setPreviewPictureURL("");
     }
 
     private com.di72nn.stuff.wallabag.apiwrapper.models.Tag findApiTagByID(

